### ISSUE TITLE
doctest --split is broken

### ISSUE THREAD

**Issue by jlherren** [sympy/sympy#19141]
On current master, running `bin/doctest` prints a few deprecation warnings, but completes with exit status 0. But running `bin/doctest --split 35/100` fails with exit status 1, see below. Seems to be since #19006 was merged.
```
===================================================================== test process starts =====================================================================
executable:         /home/jlh/virtualenvs/sympy-dev/bin/python  (3.7.3-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       python 
numpy:              1.18.2
hash randomization: on (PYTHONHASHSEED=3537508537)
split:              35/100

sympy/printing/mathml.py[1] .                                                                                                                              [OK]
sympy/printing/rcode.py[1] .                                                                                                                               [OK]
sympy/printing/llvmjitcode.py[1] s                                                                                                                         [OK]
sympy/printing/tableform.py[4] ..F.                                                                                                                      [FAIL]
sympy/printing/ccode.py[1] .                                                                                                                               [OK]
sympy/printing/tree.py[2] ..                                                                                                                               [OK]

_______________________________________________________________________________________________________________________________________________________________
_________________________________________________________ sympy.printing.tableform.TableForm.__init__ _________________________________________________________
File "/<path>/sympy/sympy/printing/tableform.py", line 110, in sympy.printing.tableform.TableForm.__init__
Failed example:
    TableForm([['.'*(j if not i%2 else 1) for i in range(3)]
               for j in range(4)], alignments='rcl')
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python3.7/doctest.py", line 1329, in __run
        compileflags, 1), test.globs)
      File "<doctest sympy.printing.tableform.TableForm.__init__[3]>", line 2, in <module>
        for j in range(4)], alignments='rcl')
      File "/<path>/sympy/sympy/printing/tableform.py", line 147, in __init__
        _lines = Tuple(*data)
      File "/<path>/sympy/sympy/core/containers.py", line 51, in __new__
        obj = Basic.__new__(cls, *args)
      File "/<path>/sympy/sympy/core/containers.py", line 50, in <genexpr>
        args = (sympify(arg) for arg in args)
      File "/<path>/sympy/sympy/core/sympify.py", line 365, in sympify
        rational=rational) for x in a])
      File "/<path>/sympy/sympy/core/sympify.py", line 365, in <listcomp>
        rational=rational) for x in a])
      File "/<path>/sympy/sympy/core/sympify.py", line 389, in sympify
        deprecated_since_version='1.6'
      File "/<path>/sympy/sympy/utilities/exceptions.py", line 185, in warn
        warnings.warn(self, stacklevel=stacklevel)
    sympy.utilities.exceptions.SymPyDeprecationWarning: 

    String fallback in sympify has been deprecated since SymPy 1.6. Use
    sympify(str(obj)) or sympy.core.sympify.converter or obj._sympy_
    instead. See https://github.com/sympy/sympy/issues/18066 for more
    info.


=============================================== tests finished: 8 passed, 1 failed, 1 skipped, in 0.09 seconds ================================================
DO *NOT* COMMIT!
```

**Comment by jksuom** [comment]
`TableForm` tries to sympify all entries in the table and, if that fails, makes symbols that are printed as expected. https://github.com/sympy/sympy/blob/21fe47c6b0b39379b8f2bb00318467a3100978a7/sympy/printing/tableform.py#L141-L144
That is what should happen in the failed example: `sympify` raises on the dotstrings `'.'*(j if not i%2 else 1)` and those are transformed to symbols like `Symbol('..')`.

However, there is one case where `sympify` succeeds:
```
>>> sympify('...')
Ellipsis
```
It looks like this result is unexpected. The code does not know how to handle it and raises a warning.

**Comment by jlherren** [comment]
The issue I meant to point out is that a deprecation warning causes a hard error when `--split` is in use, but not otherwise.

`TableForm` should be fixed of course, but that's not what I intended this issue to be about. It was just an example, there are also other deprecation warnings triggered by the doctests.

**Comment by oscarbenjamin** [comment]
I think I fixed this in #19247

**Comment by jksuom** [comment]
>  I think I fixed this in #19247

I think so too. `...` will not be sympified as it is converted to a Symbol when created.
